src/price_data.py

```diff
@@ -133,6 +133,91 @@ def _get_price_binance(
         average_price = total_cost / total_quantity
         return average_price
 
+    @misc.delayed
+    def _get_price_coinbase(self, base_asset: str, utc_time: datetime.datetime, quote_asset: str, swapped_symbols: bool = False) -> decimal.Decimal:
+        # Use Coinbase Pro prices
+        return self._get_price_coinbase_pro(base_asset, utc_time, quote_asset, swapped_symbols)
+
+    @misc.delayed
+    def _get_price_coinbase_pro(self, base_asset: str, utc_time: datetime.datetime, quote_asset: str, swapped_symbols: bool = False) -> decimal.Decimal:
+        """Retrieve price from Coinbase Pro official REST API.
+
+        The price is calculated as the average price of the opening and closing price
+        around 1 minute around `utc_time`.
+
+        None existing pairs like `TWTEUR` are calculated as
+        `TWTBTC * BTCEUR`.
+
+        Documentation: https://docs.pro.coinbase.com
+
+        Args:
+            start	        Start time in ISO 8601
+            end	            End time in ISO 8601
+            granularity	    Desired timeslice in seconds (60, 300, 900, 3600, 21600 or 86400)
+
+        Raises:
+            RuntimeError: Unable to retrieve price data.
+
+        Returns:
+            Decimal: Price of asset pair.
+        """
+
+        root_url = "https://api.pro.coinbase.com"
+        symbol = f"{base_asset}-{quote_asset}"
+        startTime = misc.to_iso_timestamp(utc_time - datetime.timedelta(minutes=5))
+        endTime = misc.to_iso_timestamp(utc_time + datetime.timedelta(minutes=5))
+        sub_url = f"/products/{symbol}/candles?start={startTime}&end={endTime}&granularity=60"
+        url = root_url + sub_url
+
+        log.debug("Calling %s", url)
+        response = requests.get(url)
+        data = json.loads(response.text)
+
+        # Some combinations do not exist (e.g. `TWTEUR`), but almost anything
+        # is paired with BTC. Calculate `TWTEUR` as `TWTBTC * BTCEUR`.
+        if isinstance(data, dict) and data.get("message") == "NotFound":
+            if quote_asset == "BTC" or base_asset == "BTC":
```

Comment:
It might be cleaner, if we move this to a separate function, since we use the exact same stuff again.

Reply:
It seems that these lines are not necessary anymore since I don't use the API for Coinbase (non-Pro) anymore - so they're removed now.